modules/utils.py

```diff
@@ -323,27 +137,81 @@ def InitAutomation(args) -> None:
             log.logger(
                 "warning",
-                "No API key specified and no api.txt file found. "
+                "No vulnerability API key specified and no vuln_api.txt file found. "
                 + "Vulnerability detection is going to be slower! "
                 + "You can get your own NIST API key from "
                 + "https://nvd.nist.gov/developers/request-an-api-key",
             )
         except PermissionError:
-            log.logger("error", "Permission denied while trying to read api.txt!")
+            log.logger("error", "Permission denied while trying to read vuln_api.txt!")
 
-    return apiKey
+    if args.shodan_api:
+        shodan_api_key = args.shodan_api
+
+    else:
+        shodan_api_key = None
+        try:
+            with open("shodan_api.txt", "r", encoding="utf-8") as f:
+                shodan_api_key = f.readline().strip("\n")
+        except FileNotFoundError:
+            log.logger(
+                "warning",
+                "No Shodan API key specified and no shodan_api.txt file found. "
+                + "Shodan scanning capabilities will be disabled! "
+                + "You can get your own Shodan API key from "
+                + "https://account.shodan.io/register",
+            )
+        except PermissionError:
+            log.logger("error", "Permission denied while trying to read shodan_api.txt!")
+
+    if args.zoomeye_api:
+        zoomeye_api_key = args.zoomeye_api
+    else:
+        zoomeye_api_key = None
+        try:
+            with open("zoomeye_api.txt", "r", encoding="utf-8") as f:
+                zoomeye_api_key = f.readline().strip("\n")
+        except FileNotFoundError:
+            log.logger(
+                "warning",
+                "No ZoomEye API key specified and no zoomeye_api.txt file found. "
+                + "ZoomEye scanning capabilities will be disabled! "
+                + "You can get your own ZoomEye API key from "
+                + "https://www.zoomeye.org/api"
+            )
+        except PermissionError:
+            log.logger("error", "Permission denied while trying to read zoomeye_api.txt!")
+
+    if args.openai_api_key:
+        openai_api_key = args.openai_api_key
+    else:
+        openai_api_key = None
+        try:
+            with open("openai_api.txt", "r", encoding="utf-8") as f:
+                openai_api_key = f.readline().strip("\n")
+        except FileNotFoundError:
+            log.logger(
+                "warning",
+                "No OpenAI API key specified and no openai_api.txt file found. "
+                + "AI keyword generation capabilities will be disabled! "
+                + "You can get your own OpenAI API key from "
+                + "https://beta.openai.com/signup/",
+            )
+        except PermissionError:
+            log.logger("error", "Permission denied while trying to read openai_api.txt!")
+
+    return vuln_api_key, shodan_api_key, zoomeye_api_key, openai_api_key
 
```

Comment:
This creates too much clutter at the beginning of the program. It would be much better if these warnings show up after the banner.

Reply:
I realized they actually show up twice before and after the banner.

Reply:
Fixed